### PR TITLE
chore(metrics): expose `MetricsInterface` as return type of single metric and further improve API docs

### LINKED ISSUE #3132: Feature request: Change MetricsInterface.singleMetric() to have return type MetricsInterface
### Use case

Metrics doesn't (currently) have a good way to silence all output during unit tests (it requires configuration changes outside the scope of the unit test code.) To work around this I was going to use a fake implementation of the `MetricsInterface` interface, and pass that to actual code that generates metrics, instead of passing a reference to the concrete `Metrics` class. At unit test time I can instead pass a fake implementation of this interface that just silently swallows all requests. The problem with this workaround is that the `MetricsInterface.singleMetric()` method has to return a concrete implementation of `Metrics`.

### Solution/User Experience

Change `MetricsInterface.singleMetric()` to return `MetricsInterface` . I don't think (?) this should break anything, and will allow me to replace the concrete implementation at unit test time.

### Alternative solutions

```Markdown
Add functionality to the MetricsOptions interface (used in the Metrics class constructor) to take an option something like "silent" which tells the implementation not to log. Or allow passing a custom output sink. I think these are both better options long term, but my suggestion I think is a one-liner that probably fits better with the design anyway (it's typically best to not have interfaces have types that return concretions).
```

### Acknowledgment

- [X] This feature request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR started out with the goal of modifying the return type of the `Metrics.singleMetric()` method to use `MetricsInterface` instead of `Metrics`. This small change allows for easier customization of the utility and avoids a cyclical type reference.

While working on this, I realized that the `MetricsInterface` was out of date and had no documentation whatsoever, which means its API page is empty and there's no contextual documentation in the IDE when customers hover any of the interface or properties.

At that point the goal shifted towards making sure the methods and their API docs in the `Metrics` class and the `MetricsInterface` were aligned. In doing so, I also noticed that some of the public methods were somewhat under documented and at least in some cases, there were opportunities to either clarify the purpose of the method or add a code snippet.

This ended becoming a seizable PR that further improved on the work started by @am29d in a previous PR and complemented the API docs added in that PR by expanding some of the descriptions and introducing a bunch of links between methods/sections.

For example, now the API docs for the `addMetric()` method goes in detail on how to use the method but also links to other relevant parts of the API docs:

image

This work was enabled by @am29d's work on fixing our TypeDoc setup and making sure links are properly rendered & followed. I have applied this type of pattern across the entire Metrics package and also made sure all code samples and wording is consistent across API docs & README.

Finally, while there, I noticed that we had a public method named `throwOnEmptyMetrics()` which was severely under documented and that didn't do what I'd have expected it to do. I marked it as deprecated in favor of a newly added method `setThrowOnEmtpymetrics(enabled: boolean)` which is clearer on its purpose and is also aligned with other similar public setter methods in terms of naming.

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #3132

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
The 2 accepted issues marked by SonarCloud are related to the deprecated method still being used in the unit tests. For now I marked them as such since these tests will be refactored soon when moving the utility to Vitest, and the method will eventually be removed in the next major version.

File: packages/metrics/src/Metrics.ts [hunk 1]
```diff
@@ -240,38 +276,37 @@ class Metrics extends Utility implements MetricsInterface {
...
    *
-   * By default, metrics are buffered and flushed at the end of the Lambda invocation
-   * or when calling {@link Metrics.publishStoredMetrics}.
+   * By default, metrics are buffered and flushed when calling {@link Metrics.publishStoredMetrics | `publishStoredMetrics()`} method,
+   * or at the end of the handler function when using the {@link Metrics.logMetrics | `logMetrics()`} decorator or the Middy.js middleware.
    *
-   * You can add a metric by specifying the metric name, unit, and value. For convenience,
-   * we provide a set of constants for the most common units in {@link MetricUnit}.
+   * Metrics are emitted to standard output in the Amazon CloudWatch EMF (Embedded Metric Format) schema. In AWS Lambda, the logs are
+   * automatically picked up by CloudWatch logs and processed asynchronously.
```
Comment (leandrodamascena) [inline packages/metrics/src/Metrics.ts:283]:
```suggestion
   * Metrics are emitted to standard output in the Amazon CloudWatch EMF (Embedded Metric Format) schema.
```

Customers can remove CloudWatch permissions, can subscribe logs to api or other. So, it's better to keep that this utility just emits logs in this format.

File: packages/metrics/src/Metrics.ts [hunk 2]
```diff
@@ -729,12 +887,12 @@ class Metrics extends Utility implements MetricsInterface {
    * Store a metric in the buffer.
    *
    * If the buffer is full, or the metric reaches the maximum number of values,
-   * the buffer is published to stdout.
+   * the metrics are published.
```
Comment (leandrodamascena) [inline packages/metrics/src/Metrics.ts:890]:
```suggestion
   * the metrics are flushed to stdout.
```

**Review by leandrodamascena — DISMISSED** [review]
Hey @dreamorosi! I just have 2 small comments.

I saw you're changing code in `Metrics.ts` and `types/Metrics.ts` and for me looks good. If the tests are green, good to merge.

**Review by leandrodamascena — APPROVED** [review]

**Pull request merged**